### PR TITLE
core/services/vrf: flat fee for nocancel coord v2

### PR DESCRIPTION
**Pull request by makramkd**
In addition, refactor the test suite so that we run some of the same tests for NoCancelVRFCoordinatorV2 and VRFCoordinatorV2.

### CODE REVIEW

File: contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol [hunk 1]
```diff
@@ -610,26 +608,18 @@ contract NoCancelVRFCoordinatorV2 is
...
-    uint256 paymentNoFee = (1e18 * weiPerUnitGas * (gasAfterPaymentCalculation + startGas - gasleft())) /
-      uint256(weiPerUnitLink);
     uint256 fee = 1e12 * uint256(fulfillmentFlatFeeLinkPPM);
-    if (paymentNoFee > (1e27 - fee)) {
-      revert PaymentTooLarge(); // Payment + fee cannot be more than all of the link in existence.
+    if (fee > (1e27 - fee)) {
+      revert PaymentTooLarge(); // Fee cannot be more than all of the link in existence.
     }
-    return uint96(paymentNoFee + fee);
+    return uint96(fee);
```
Comment (makramkd) [inline contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol:622]: Should we just axe this function? Is the `if (fee > (1e27 - fee)) {` check worth the gas if `fee` is always just the premium in juels?

Comment (jinhoonbang) [reply contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol:622]: I guess the validation would be more appropriate to do in `setConfig`. Also, do we know if total supply of dummy link will be 10^9?

**Review by jinhoonbang — APPROVED** [review]

File: contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol (hunk 1 above)
Comment (makramkd) [reply contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol:622]: Yeah, I believe the LINK supply on the appchains will be the same as the main chains.

Comment (makramkd) [reply contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol:622]: Good note on adding validation in `setConfig` though.

**Pull request merged**